woocommerce/woocommerce-ios
# Add Product Variations Attributes and Options Cell UI

Closes #3519 

## Description

This PR adds an **Attributes** cell to the Product Variation detail screen, behind a new feature flag for editing product variation attributes (enabled for local development/alpha only). For now, the Attributes cell doesn't do anything when tapped and always displays the options text "Any Color - Any Material."

## Changes

* Adds the `editProductVariationAttributes` feature flag set to `localDeveloper` and `alpha` builds.
* Adds the `customize` icon from gridicons (for the attributes cell icon), and an icon test case to make sure the icon is not nil.
* In `DefaultProductFormTableViewModel`, configures the attributes row. For now the cell displays static detail text (for the attribute options), and the cell is set to expand if that detail text doesn't fit on a single line.
* In `ProductVariationFormActionsFactory`, sets the attributes row to be displayed based on the feature flag. When enabled, the attributes row is displayed below the price settings/warning row and above the variation status (Enabled) row.

### Screenshots

/|Before|After
-|-|-
Variation with price|![Simulator Screen Shot - iPhone 12 mini - 2021-02-01 at 15 33 56](https://user-images.githubusercontent.com/8658164/106487152-4b91db00-64aa-11eb-8391-54fc7956abf2.png)|![Simulator Screen Shot - iPhone 12 mini - 2021-02-01 at 15 29 36](https://user-images.githubusercontent.com/8658164/106487232-606e6e80-64aa-11eb-8a81-a8039092dbb4.png)
Variation without price|![Simulator Screen Shot - iPhone 12 mini - 2021-02-01 at 15 34 00](https://user-images.githubusercontent.com/8658164/106487254-65332280-64aa-11eb-9ab3-f4ad15d00dad.png)|![Simulator Screen Shot - iPhone 12 mini - 2021-02-01 at 15 29 31](https://user-images.githubusercontent.com/8658164/106487266-66644f80-64aa-11eb-9bc3-36d040ca2301.png)





## Testing

1. Open a variable product that has existing variations.
2. Tap on Variations.
3. Select a variation that has a price set.
4. Confirm you see the Attributes cell below the Price cell in the variation details.
5. Go back to the variations list.
6. Select a variation with no price set.
7. Confirm you see the Attributes cell below the "Variations without price ..." warning.
8. In `DefaultFeatureFlagService`, set `.editProductVariationAttributes` to false and confirm no Attributes cell appears in the product variation details.

## Submitter Checklist

Update release notes:
